mixer/adapter/solarwinds/config/config.proto

```diff
@@ -17,33 +17,54 @@ syntax = "proto3";
 package adapter.solarwinds.config;
 
 import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
 
 option go_package = "config";
 option (gogoproto.goproto_getters_all) = false;
 option (gogoproto.equal_all) = false;
 option (gogoproto.gostring_all) = false;
 
 message Params {
-    
+    // AppOptics Access Token needed to send metrics to AppOptics. If no access token is given then metrics
+    // will NOT be shipped to AppOptics
     string appoptics_access_token = 1;
-    
-    string papertrail_url = 2;
 
-    string papertrail_local_retention = 3;
-    
+    // Optional. Max batch size of metrics to be sent to AppOptics.
+    // AppOptics does not allow batch size greater than 1000.
+    // If this is not specified a default batch size of 1000 will be used.
+    int32 appoptics_batch_size = 2;
+
+    // Papertrail url to ship logs to. If no papertrail url is given then the logs will NOT be shipped but rather
+    // dropped.
+    string papertrail_url = 3;
+
+    // This is the duration for which logs will be persisted locally until it is shipped to papertrail in the event
+    // of a network failure
+    google.protobuf.Duration papertrail_local_retention_duration = 4 [(gogoproto.nullable)=true, (gogoproto.stdduration) = true];
+
+    // Describes how to represent an Istio metric in Solarwinds AppOptics
     message MetricInfo {
+        // The name is used to register the metric.
         string name = 1;
-        string instance_name = 2;
-        string description = 3;
-        repeated string label_names = 4;
+
+        // Optional. A human readable description of this metric.
+        string description = 2;
+
+        // The names of labels to use: these need to match the dimensions of the Istio metric.
```

Comment:
It is possible to validate this inside the Validate func. The `builder` is passed both the adapter configuration as well as the types via the `SetType func`. The passed in types contains information about all the instances that will be dispatched to the adapter, therefore it is possible to validate if the labels here match the dimensions in the metric.

Reply:
Also validate if the instance_name is one among'st the available instance names.

Reply:
Great suggestion. I have tried to incorporate this. Please let me know your thoughts.